src/Enable.Extensions.Queuing.AzureServiceBus/Internal/AzureServiceBusQueueClient.cs

```diff
@@ -123,19 +130,36 @@ protected override void Dispose(bool disposing)
             {
                 if (disposing)
                 {
-                    _messageReceiver.CloseAsync()
-                        .GetAwaiter()
-                        .GetResult();
-
-                    _messageSender.CloseAsync()
-                        .GetAwaiter()
-                        .GetResult();
+                    // Use Monitor to prevent race conditions
+                    Monitor.Enter(_queues);
+                    try
+                    {
+                        var refCount = _queue.RemoveReference();
+                        if (refCount == 0)
+                        {
+                            var removed = _queues.TryRemove(_queueKey, out _);
+                            if (removed)
+                            {
+                                _queue.Dispose();
+                            }
+                            else
+                            {
+                                // This case should never occur, because we used Monitor to avoid race conditions
+                                // Raise an exception if this occurs to notify us that this code is probably wrong
```

Comment:
Let's wrap these comments at 80 characters.

Reply:
The code doesn't wrap at 80 chars and that is a pretty low character limit. I can do it if you like but it doesn't look elegant (this comment becomes 4 lines).

Reply:
Done.